src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,120 @@
+import * as ld from "lodash";
+
+// First Group will be the variable value. Doesn't fetch the variable names
+const prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+
+export class PrologResult {
+	private readonly rawResults: string[];
+	private readonly results: Map<string, Map<number, string|boolean>>;
+
+	constructor(rawResults: string[]) {
+		this.rawResults = rawResults;
+		this.results = new Map();
+	}
+
+	public getResults(): Map<string, Map<number, string|boolean>> {
+		if (this.results.size !== 0) { return this.results; }
+
+		const results = new Map();
+		let i: number;
+
+		this.getVariables().forEach((e: string) => {
+			results.set(e, new Map());
+			i = 0;
+
+			this.getResultsFor(e).forEach((e2: string) => {
+				results.get(e).set(i, e2);
+				i++;
+			});
+
+		});
+
+		results.set("Boolean", new Map());
+		i = 0;
+
+		this.getBooleans().forEach((e: boolean) => {
+			results.get("Boolean").set(i, e);
+			i++;
+		});
+
+		return results;
```

Comment:
Shouldn't you also set `this.results = results` (or copy all the values from `result` into `this.result`)? Otherwise `this.results` will always be an empty map.